File: CustomCADs.Domain/Entities/Order.cs
using CustomCADs.Domain.Enums;

namespace CustomCADs.Domain.Entities
{
    public class Order 
    {
        public int Id { get; set; }
        public required string Name { get; set; }
        public required string Description { get; set; } 
        public DateTime OrderDate { get; set; }
        public OrderStatus Status { get; set; }
        public bool ShouldBeDelivered { get; set; }
        public required string ImagePath { get; set; } 
        public int CategoryId { get; set; }
        public required Category Category { get; set; } 
        public required string BuyerId { get; set; } 
        public required User Buyer { get; set; } 
        public string? DesignerId { get; set; }
        public User? Designer { get; set; } 
        public int? CadId { get; set; }
        public Cad? Cad { get; set; }        
    }
}

File: customcads/src/interfaces/order.ts
import ICategory from "./category"

export default interface IOrder { 
    id: number
    shouldBeDelivered: boolean
    name: string
    description: string
    status: string
    orderDate: string
    imagePath: string
    designerName?: string
    designerEmail?: string
    buyerName: string
    cadId?: number
    category: ICategory
}

export const emptyOrder: IOrder = {
    id: 0,
    shouldBeDelivered: false,
    name: '',
    description: '',
    status: '',
    orderDate: '',
    imagePath: '',
    buyerName: '',
    category: { id: 0, name: '' },
};